components/builder-depot/src/server.rs

```diff
@@ -100,6 +102,21 @@ impl Key for TestableBroker {
     type Value = Self;
 }
 
+pub struct RoutedMessages(HashMap<TypeId, Vec<u8>>);
+
+impl RoutedMessages {
+    pub fn get<M: Routable>(&self) -> Result<M> {
+        let msg_type = &TypeId::of::<M>();
+        match self.0.get(msg_type) {
+            Some(msg) => {
+                Ok(parse_from_bytes::<M>(msg).expect(&format!("Unable to parse {:?} message",
```

Comment:
Same thing here - we panic if we get bad messages on the wire, which seems brittle.